source/common/http/http2/codec_impl.h

```diff
@@ -84,6 +84,17 @@ class ConnectionImpl : public virtual Connection, Logger::Loggable<Logger::Id::h
   Protocol protocol() override { return Protocol::Http2; }
   void shutdownNotice() override;
   bool wantsToWrite() override { return nghttp2_session_want_write(session_); }
+  // Propogate network connection watermark events to each stream on the connection.
+  void onUnderlyingConnectionAboveWriteBufferHighWatermark() override {
+    for (auto& stream : active_streams_) {
+      stream->runHighWatermarkCallbacks();
+    }
+  }
+  void onUnderlyingConnectionBelowWriteBufferLowWatermark() override {
```

Comment:
We can delete these definitions below in ClientConnection now, right? I think  they are just duplicate overrides. Also, love the changer to push over watermark tracking into connection class. Much cleaner! :)